test/unit/event.js

```diff
@@ -5,20 +5,6 @@ module( "event", {
 	teardown: moduleTeardown
 });
 
-test("null or undefined handler", function() {
-	expect(2);
-	// Supports Fixes bug #7229
-	try {
-		jQuery("#firstp").on( "click", null );
-		ok(true, "Passing a null handler will not throw an exception");
-	} catch ( e ) {}
-
-	try {
-		jQuery("#firstp").on( "click", undefined );
-		ok(true, "Passing an undefined handler will not throw an exception");
-	} catch ( e ) {}
-});
-
```

Comment:
@timmywil if throws or not is undefined/undocumented behaviour, so it seems we explicitly shouldn't add test for it.

@dmethvin i guess it's better to throw right away then later, but see above.

-- 
By "undefined behaviour" i mean <a href="http://en.wikipedia.org/wiki/Undefined_behavior">term</a> from the C world, so basically we free to do whatever we want.

"Knowning" @dmethvin though, i guess you more concern of how much yelling we would get if do this. I think not a lot, but that is my perception :-)